src/test/librados/io.cc

```diff
@@ -826,6 +827,12 @@ TEST_F(LibRadosIoEC, OverlappingWriteRoundTrip) {
   char *buf = (char *)new char[dbsize];
   char *buf2 = (char *)new char[bsize];
   char *buf3 = (char *)new char[dbsize];
+  auto lam = [&] {
+            delete[] buf;
+            delete[] buf2;
+            delete[] buf3;
+          };
+  scope_guard<decltype(lam)> sg(std::forward<decltype(lam)>(lam));
```

Comment:
but i don't think we should trade the readability for appeasing the static analyzer if the code is correct. moreover, we can avoid use names like `lam` (i figured out it's the short for lambda just now).